Logging/RockLib.Logging.Analyzers/CommonExtensions.cs

```diff
@@ -10,6 +10,24 @@ namespace RockLib.Logging.Analyzers
 {
     public static class CommonExtensions
     {
+        public static bool IsException(this ITypeSymbol typeSymbol)
```

Comment:
If you have access to a `Compilation` object, it has a method called `ClassifyCommonConversion()`. You can do a check against the `Exception` symbol (assuming you've loaded it with `GetTypeByMetadataName()`). Then you might be able to do something like:
```
if(compilation.ClassifyCommonConversion(typeSymbol, exceptionSymbol).IsImplicit)
```
And that will tell you if the type derives from the `Exception` type.

Reply:
`Compilation` is a bit higher up the call stack, but let me see what I can do.

Reply:
Understood. If you can, I'd see if you can pull in the type symbol for `Exception`, rather than a string comparison. In this case, you're probably safe because your context is just dealing with exceptions, so it's **highly** unlikely someone would create an `Exception` that derives from `System.Exception`, because....well, that would be evil :). But comparing with type symbols is a safer comparison.